core/plugin_api.php

```diff
@@ -190,10 +190,19 @@ function plugin_route_group( $p_base_name = null ) {
  * @param string $p_base_name Plugin base name.
  * @return mixed File path or false if FNF
  */
-function plugin_file_path( $p_filename, $p_base_name ) {
+function plugin_file_path( $p_filename = null, $p_base_name = null ) {
 	$t_file_path = config_get_global( 'plugin_path' );
-	$t_file_path .= $p_base_name . DIRECTORY_SEPARATOR;
-	$t_file_path .= 'files' . DIRECTORY_SEPARATOR . $p_filename;
+	if( is_null( $p_base_name ) ) {
+		$t_file_path .= plugin_get_current() . DIRECTORY_SEPARATOR;
+	} else {
+		$t_file_path .= $p_base_name . DIRECTORY_SEPARATOR;
+	}
+	$t_file_path .= 'files' . DIRECTORY_SEPARATOR;
```

Comment:
Use a short ternary `$p_base_name ?: plugin_get_current()` instead of an il...else block.

Also, I'd rather have a single assignment with several concats, on  multiple lines if needed to avoid long lines, than a series of $var .= xxx as in the original code - much easier to read IMHO.

```php
$t_file_path = config_get_global( 'plugin_path' )
    . ( $p_base_name ?: plugin_get_current() ) . DIRECTORY_SEPARATOR
    . 'files' . DIRECTORY_SEPARATOR;